app/knight.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(
+            self,
+            knight: dict
+    ) -> None:
+        self.name: str = knight.get("name")
+        self.power: int = knight.get("power")
+        self.hp: int = knight.get("hp")
+        self.armour: list[dict] = knight.get("armour")
+        self.weapon: dict = knight.get("weapon")
+        self.potion = knight.get("potion")
+        self.protection = 0
+
+    def apply_equipment(self) -> None:
+        self.protection = sum(
+            [
+                armour.get("protection")
+                for armour
+                in self.armour
+            ]
+        )

```

Comment:
no need to create list
```suggestion
        self.protection = sum(
            armour.get("protection")
            for armour in self.armour
        )

```